### PR TITLE
Extract builders and javadoc

### PR DESCRIPTION
**Pull request by MechaRyRy**
## Problem
I discovered it was quite confusing working with the `Batch.Builders` because of the nested classes and interfaces 😬

## Solution
Pull up the builders and javadoc to avoid confusion.

### CODE REVIEW

File: library/src/main/java/com/novoda/downloadmanager/BatchFileBuilder.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+/**
+ * Adds {@link BatchFile} to the parent {@link Batch} using the {@link BatchBuilder}
+ * and {@link BatchFileBuilder} fluent APIs.
+ */
+public interface BatchFileBuilder {
+
+    /**
+     * Sets {@link BatchFileBuilder} to build a {@link BatchFile} with a corresponding
+     * {@param downloadFileId} that can be used to lookup a download file belonging to
```
Comment (danybony) [inline library/src/main/java/com/novoda/downloadmanager/BatchFileBuilder.java]: What about replacing 
`that can be used to lookup a download file`
with
`that can be later used to retrieve a downloaded file`?

File: library/src/main/java/com/novoda/downloadmanager/BatchFileBuilder.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+     * a given {@link Batch}.
+     *
+     * @param downloadFileId to flag a {@link BatchFile} with.
+     * @return {@link BatchFileBuilder}.
+     */
+    BatchFileBuilder withIdentifier(DownloadFileId downloadFileId);
+
+    /**
+     * Sets {@link BatchFileBuilder} to build a {@link BatchFile} that will be saved
+     * to the given {@param path}.
```
Comment (danybony) [inline library/src/main/java/com/novoda/downloadmanager/BatchFileBuilder.java]: Maybe specify here that the path is only referring to the directory that will contain the file.
And also, is it relative or absolute?

File: library/src/main/java/com/novoda/downloadmanager/LiteBatchBuilder.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        this.title = title;
+        this.batchFiles = batchFiles;
+    }
+
+    @Override
+    public void withFile(BatchFile batchFile) {
+        batchFiles.add(batchFile);
+    }
+
+    private BatchFileBuilder fileBuilder;
```
Comment (danybony) [inline library/src/main/java/com/novoda/downloadmanager/LiteBatchBuilder.java]: Field before the constructor

**Review by zegnus — APPROVED** [review]

**Review by danybony — COMMENTED** [review]
💯

**Pull request merged**